oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticPropertyDefinition.java

```diff
@@ -50,6 +61,14 @@ public ElasticPropertyDefinition(IndexDefinition.IndexingRule idxDefn, String no
                     getOptionalValue(defn, PROP_CANDIDATES, DEFAULT_CANDIDATES));
         }
         this.useInFullTextQuery = this.dynamicBoost && getOptionalValue(defn, PROP_USE_IN_FULL_TEXT_QUERY, true);
+        boolean flattened = getOptionalValue(defn, PROP_IS_FLATTENED, PROP_IS_FLATTENED_DEFAULT);
+        if (analyzed) {
```

Comment:
I am wondering if we should actually fail in this case.
If we don't, the index won't be created as flattened and it might fail while indexing in case the number of fields exceeds the maximum.

Reply:
> I am wondering if we should actually fail in this case.

Sorry I saw the comment too late...

Yes, we could reject this combination. The main reason I didn't is: If we change the default, this would immediately fail many tests. I wanted to be able to test with the setting enabled by default... Sure, there are other options as well: for example, only fail if analyzed is enabled and flattened is explicitly set.